targets/generic-x86_64/flake-module.nix

```diff
@@ -2,117 +2,113 @@
+                  }
+                ];
               };
+            };
 
-              host.networking.enable = true;
+            host.networking.enable = true;
 
-              # Enable all the default UI applications
-              profiles = {
-                applications.enable = true;
-                release.enable = variant == "release";
-                debug.enable = variant == "debug";
-                # Uncomment this line to use Labwc instead of Weston:
-                #graphics.compositor = "labwc";
-              };
-              reference.programs.windows-launcher.enable = true;
+            # Enable all the default UI applications
+            profiles = {
+              applications.enable = true;
+              release.enable = lib.mkDefault true;
+              debug.enable = lib.mkDefault false;
+              # Uncomment this line to use Labwc instead of Weston:
+              #graphics.compositor = "labwc";
             };
+            reference.programs.windows-launcher.enable = true;
+          };
 
-            #TODO: how to handle the majority of laptops that need a little
-            # something extra?
-            # SEE: https://github.com/NixOS/nixos-hardware/blob/master/flake.nix
-            # nixos-hardware.nixosModules.lenovo-thinkpad-x1-10th-gen
+          #TODO: how to handle the majority of laptops that need a little
+          # something extra?
+          # SEE: https://github.com/NixOS/nixos-hardware/blob/master/flake.nix
+          # nixos-hardware.nixosModules.lenovo-thinkpad-x1-10th-gen
 
-            boot.kernelParams = [
-              "intel_iommu=on,igx_off,sm_on"
-              "iommu=pt"
+          boot.kernelParams = [
+            "intel_iommu=on,igx_off,sm_on"
+            "iommu=pt"
 
-              # TODO: Change per your device
-              # Passthrough Intel WiFi card
-              "vfio-pci.ids=8086:a0f0"
-            ];
-          }
-        ] ++ extraModules;
+            # TODO: Change per your device
+            # Passthrough Intel WiFi card
+            "vfio-pci.ids=8086:a0f0"
+          ];
+        }
+      ];
+    };
+    nixosModules.generic-x86-debug = {
+      ghaf.development.usb-serial.enable = true;
+      profiles = {
+        applications.enable = true;
+        release.enable = false;
+        debug.enable = true;
       };
-    in
-    {
-      inherit hostConfiguration;
-      name = "${name}-${variant}";
-      package = hostConfiguration.config.system.build.${hostConfiguration.config.formatAttr};
     };
-  debugModules = [ { ghaf.development.usb-serial.enable = true; } ];
-  targets = [
-    (generic-x86 "debug" debugModules)
-    (generic-x86 "release" [ ])
-  ];
-in
-{
-  flake = {
-    nixosConfigurations = builtins.listToAttrs (
-      map (t: lib.nameValuePair t.name t.hostConfiguration) targets
-    );
+    nixosConfigurations = {
+      generic-x86_64-release = lib.nixosSystem { modules = [ self.nixosModules.generic-x86-release ]; };
```

Comment:
In theory this could be done more dry, but the code needed for that seemed longer and more complex, so I didn't.